src/support/cli.ts

```diff
@@ -28,14 +28,53 @@ function formatLine (color: Function, type: string, line: string, prefix?: strin
 }
 
 /**
+ * Available log levels.
+ */
+let logLevels: LogLevels = {
+    debug: 0,
+    info: 1,
+    warn: 2,
+    error: 3,
+    silent: 4
+}
+interface LogLevels {
+    debug: number
+    info: number
+    warn: number
+    error: number
+    silent: number
+    [key: string]: number
+}
+
+/**
+ * Current log level.  Defaults to emit all.
+ */
+let logLevel: number = logLevels.info
+
+/**
+ * Set the level of logs to emit.
+ */
+export function setLogLevel(level: string): number {
+    let match = logLevels[level]
+    if (!match && match !== logLevels.debug) {
```

Comment:
Please don't do checks like this, it's counter productive to fight JavaScript. If you wanted to do this sort of check, you could have done `match != null`. However, this doesn't handle prototype properties. Prefer:

```ts
function setLogLevel (level: string): number {
  if (!logLevels.hasOwnProperty(level)) {
    logError(`Invalid log level, available options are ${listify(Object.keys(logLevels))}`)
    return
  }

  return (logLevel = logLevels[level])
}